### ISSUE TITLE
Auto-install maybe didn't work?

### ISSUE THREAD

**Issue by Abisco** [s0t7x/decky-sunshine#14]
Not sure what's going on, I've never used sunshine before but saw your plugin and figured I'd give it a whirl. I have the plugin installed via decky, and have added the moonlight clients on both my iPad and my Mac. When enabling sunshine via decky, sometimes it gives the login prompt and sometimes it just shows the pinpad, regardless I eventually get to the pin pad page.

Moonlight on both my devices require adding the device via the IP address, once added it shows 'steamdeck' in the devices list. I click on it and it gives the key. I enter it in my steam deck but then... nothing happens. Ive tried multiple times, restarting my deck, restarting the apps on my ipad and computer (which note, it seems to think the device is permanently asleep and I have to remove the device then readd it).

Love your plug-in idea but definitely something is up for my decks installation.
Thanks so much for your hard work! Hope this issue gets resolved :)

**Comment by s0t7x** [comment]
I'll definitely take a closer look into what might be causing this behavior.

To better understand what's going on, it would be helpful if you could provide the log files from decky-sunshine. This will allow me to dive deeper into your error and pinpoint where the issue might be arising from.

The log files are located on your steamdeck at:
`/home/deck/homebrew/logs/Decky Sunshine`

In the meantime, I'll also try to replicate the process on another Steam Deck as soon as I can get my hands on a second one. While Sunshine is working smoothly on my own device, I had the software installed previously and removed it to allow the plugin to install it anew, which might have contributed to its successful functioning.

Thank you for your patience and for bringing this to my attention.

**Comment by s0t7x** [comment]
Maybe related to #10

**Comment by stavfx** [comment]
Something similar (or the same thing) happened to me and it was my own fault.
When first running the Plugin, I saw the message "you must login to the web ui first" and a "Login" button and assumed that it'll take me to complete whatever setup steps are required.
So essentialy I didn't go to the web ui for the initial setup of sunshine credentials, but when I entered a username/password in the Plugin ui, I still got thru to the pin pad, so I thought I had set a new password successfully and I was ready to go.

TLDR:
You must go to https://<your-deck-ip>:47990 to set up a username and password in the browser *before* you try to login via the Plugin.

More notes:
The initial setup doesn't have to be done on the Deck itself, you can use any device on the same network as long as you know your Deck's IP.
If you do prefer to use a browser on the deck itself, this will also work: https://localhost:47990

**Comment by stavfx** [comment]
I think the main issue here is that providing any credentials while credential setup isn't complete - appears like a successful login (pin pad appears). 
I'm not sure that's exactly what is happening in https://github.com/s0t7x/decky-sunshine/issues/10, but maybe.

**Comment by Abisco** [comment]
> Something similar (or the same thing) happened to me and it was my own fault. When first running the Plugin, I saw the message "you must login to the web ui first" and a "Login" button and assumed that it'll take me to complete whatever setup steps are required. So essentialy I didn't go to the web ui for the initial setup of sunshine credentials, but when I entered a username/password in the Plugin ui, I still got thru to the pin pad, so I thought I had set a new password successfully and I was ready to go.
> 
> TLDR: You must go to https://<_your-deck-ip_>:47990 to set up a username and password in the browser _before_ you try to login via the Plugin.
> 
> More notes: The initial setup doesn't have to be done on the Deck itself, you can use any device on the same network as long as you know your Deck's IP. If you do prefer to use a browser on the deck itself, this will also work: https://localhost:47990

For this, going to the IP in desktop mode doesn't seem to work (I assume that the sunshine decky plugin enabled sunshine only in game mode), but ill try the ip via another device on my network. 

FWIW I think maybe I saw that login button and thought pressing login would take me to complete the setup as well, but nothing happened.

Also @s0t7x Reading the logs, it seems to say `PIN send False` whenever I enter in the pin (though I'm 100% sure I did it right and even tried multiple times before). So must be related to the above. Will try the fix and report back

**Comment by s0t7x** [comment]
I would consider this a workaround as I was aware of the initial setup and thought I've dealt with it after the first Installation.
See [https://github.com/s0t7x/decky-sunshine/blob/af007d5c593393029250a257170a844a9e36d763/main.py#L80](Line 80)
`                self.sunshineController.setUser("decky_sunshine", "decky_sunshine", "decky_sunshine")`

There seems to be something odd happening.

**Comment by Abisco** [comment]
Okay confirmed that visiting the URL above on another computer (https://<theipofmydeck>:47990 ) shows the signup page, so something def is wrong with the setUser code

**Comment by stavfx** [comment]
Oh I wasn't aware that it was supposed to set up creds automatically. Yeah that definitely didn't work on my setup either
